src/+otp/+trigonometricdae/TrigonometricDAEProblem.m

```diff
@@ -13,6 +13,14 @@ function onSettingsChanged(obj)
                 'MassSingular', 'yes', ...
                 'Vectorized', 'on');
         end
+        
+        function y = internalSolveExactly(obj, t)
+            if ~isequal(obj.Y0, [sinh(0.5); tanh(0.5)])
```

Comment:
Are we sure this will always work, maybe make sure it is within some epsilon?

Reply:
This will probably fail for non-double data types, but at least it is always conservative. If the user never modifies the IC from default it should be fine. I hesitate to allow tolerances cause you can always manufacture edge cases where things break.